File: app/models/local.py
from datetime import datetime
from typing import Optional, List
from sqlmodel import Field, SQLModel, Relationship

class Local(SQLModel, table=True):
    __tablename__ = "locais"

    id: Optional[int] = Field(default=None, primary_key=True)
    local_id: str = Field(index=True, unique=True, description="S-012, L-05")
    tipo_local: str = Field(default="SALA", description="SALA or LAB")
    sala: Optional[str] = None
    predio: Optional[int] = None
    andar: Optional[int] = None
    tipo_ambiente: Optional[str] = None
    status: Optional[str] = None
    observacao: Optional[str] = None
    setor: Optional[str] = None
    
    # Infra fields
    quantidade_projetores: int = Field(default=0)
    saida_projetor: Optional[str] = None
    tomada_padrao: Optional[str] = Field(default=None, description="ANTIGO or NOVO")
    adaptador_dp_vga: int = Field(default=0)
    adaptador_dp_hdmi: int = Field(default=0)
    adaptador_hdmi_vga: int = Field(default=0)
    adaptador_duplicador_vga: int = Field(default=0)
    adaptador_outros: Optional[str] = None
    
    ultimo_responsavel: Optional[str] = None
    ultimo_contato: Optional[str] = None
    
    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime = Field(default_factory=datetime.utcnow)
    
    maquinas: List["Maquina"] = Relationship(back_populates="local")
    historicos: List["Historico"] = Relationship(back_populates="local")

File: web/src/types/local.ts
export interface LocalBase {
  local_id: string;
  tipo_local: "SALA" | "LAB" | string;
  sala?: string;
  predio?: number;
  andar?: number;
  tipo_ambiente?: string;
  status?: string;
  observacao?: string;
  setor?: string;
  
  quantidade_projetores: number;
  saida_projetor?: string;
  tomada_padrao?: "ANTIGO" | "NOVO" | string;
  adaptador_dp_vga: number;
  adaptador_dp_hdmi: number;
  adaptador_hdmi_vga: number;
  adaptador_duplicador_vga: number;
  adaptador_outros?: string;
  
  ultimo_responsavel?: string;
  ultimo_contato?: string;
}

export type LocalCreate = LocalBase;

export type LocalUpdate = Partial<LocalBase>;

export interface LocalRead extends LocalBase {
  id: number;
  created_at: string;
  updated_at: string;
}

export interface MaquinaBase {
  modelo: string;
  quantidade: number;
  processador?: string;
  propriedade?: "PROPRIO" | "ALUGADO" | string;
  ram_gb?: number;
  ram_modelo?: string;
  ram_tipo?: string;
  armazenamento_gb?: number;
  armazenamento_modelo?: string;
  armazenamento_tipo?: "HDD" | "SSD_SATA" | "SSD_NVME" | string;
  video_dp: number;
  video_hdmi: number;
  video_vga: number;
}

export type MaquinaCreate = MaquinaBase;

export type MaquinaUpdate = Partial<MaquinaBase>;

export interface MaquinaRead extends MaquinaBase {
  id: number;
  local_ref_id: number;
  created_at: string;
  updated_at: string;
}

export interface LocalDetailRead {
  local: LocalRead;
  maquinas: MaquinaRead[];
}
